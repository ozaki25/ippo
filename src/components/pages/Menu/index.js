import React from 'react';
import { Badge, Tab, Tabs } from '@material-ui/core';
import { AddBoxRounded, HomeRounded, NotificationsRounded } from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';
import styled from 'styled-components';
import propTypes from 'prop-types';
import OverlaySpinner from 'src/components/molecules/OverlaySpinner';
import EventsOverview from 'src/components/organisms/EventsOverview';
import EventCreateForm from 'src/components/organisms/EventCreateForm';
import NotificationList from 'src/components/organisms/NotificationList';
import Container from 'src/components/templates/Container';
import ROUTES from 'src/constants/routes';
import MENU_ITEMS from 'src/constants/menuItems';

const styles = theme => ({
  bottomBar: {
    backgroundColor: theme.palette.primary[50],
    bottom: 0,
    position: 'fixed',
    width: '100%',
    zIndex: 1300,
  },
});

const ContainerWithTabs = styled.div`
  margin-bottom: 48px;
`;

class Menu extends React.Component {
  componentDidMount() {
    this.props.data.refetch();
  }

  handleChange = (event, value) =>
    this.props.history.replace(`${ROUTES.Menu}?tab=${MENU_ITEMS.findItemByValue(value).title}`);

  onSubmitCreateEvent = async event => {
    const {
      createEvent,
      authUser: { uid, displayName },
      history,
    } = this.props;
    const result = await createEvent({
      variables: { event: { ...event, uid, name: displayName } },
    });
    history.push(`${ROUTES.Tweets}?hashtag=${event.hashtag}`);
    return result;
  };

  render() {
    const {
      data: { allEvents, loading },
      user: { fetchUser },
      tab,
      classes,
      authUser,
      history,
      firebase,
    } = this.props;
    const item = MENU_ITEMS.findItemByTitle(tab);
    const value = item ? item.value : MENU_ITEMS.HOME.value;
    const uncheckedNotificationCount = fetchUser
      ? fetchUser.notifications.filter(n => !n.checked).length
      : 0;
    return (
      <>
        <ContainerWithTabs>
          <Container
            title={MENU_ITEMS.findItemByValue(value).title}
            authUser={authUser}
            history={history}
            firebase={firebase}
            noPadding={[MENU_ITEMS.HOME.value, MENU_ITEMS.NOTIFICATION.value].includes(value)}
          >
            {value === MENU_ITEMS.HOME.value &&
              (loading ? (
                <OverlaySpinner visible={loading} />
              ) : (
                <EventsOverview
                  joined={allEvents.joined}
                  organized={allEvents.organized}
                  recommended={allEvents.recommended}
                  internal={allEvents.internal}
                  external={allEvents.external}
                  history={history}
                />
              ))}
            {value === MENU_ITEMS.NEW_EVENT.value && (
              <EventCreateForm onSubmit={this.onSubmitCreateEvent} />
            )}
            {value === MENU_ITEMS.NOTIFICATION.value && (
              <NotificationList
                history={history}
                notifications={fetchUser ? fetchUser.notifications : []}
              />
            )}
          </Container>
        </ContainerWithTabs>
        <Tabs
          value={value}
          onChange={this.handleChange}
          className={classes.bottomBar}
          variant="fullWidth"
          indicatorColor="primary"
          textColor="primary"
        >
          <Tab icon={<HomeRounded />} />
          <Tab icon={<AddBoxRounded />} />
          <Tab
            icon={
              <Badge
                className={classes.badge}
                badgeContent={uncheckedNotificationCount}
                color="primary"
              >
                <NotificationsRounded />
              </Badge>
            }
          />
        </Tabs>
      </>
    );
  }
}

Menu.displayName = 'Menu';

Menu.propTypes = {
  data: propTypes.shape({
    loading: propTypes.bool.isRequired,
    allEvents: propTypes.shape({
      joined: propTypes.arrayOf(
        propTypes.shape({
          id: propTypes.string,
          title: propTypes.string,
          catchMessage: propTypes.string,
          place: propTypes.string,
          startedAt: propTypes.string,
        }),
      ),
      recommended: propTypes.arrayOf(
        propTypes.shape({
          id: propTypes.string,
          title: propTypes.string,
          catchMessage: propTypes.string,
          place: propTypes.string,
          startedAt: propTypes.string,
        }),
      ),
      internal: propTypes.arrayOf(
        propTypes.shape({
          id: propTypes.string,
          title: propTypes.string,
          catchMessage: propTypes.string,
          place: propTypes.string,
          startedAt: propTypes.string,
        }),
      ),
      external: propTypes.arrayOf(
        propTypes.shape({
          id: propTypes.string,
          title: propTypes.string,
          eventUrl: propTypes.string,
          catchMessage: propTypes.string,
          place: propTypes.string,
          startedAt: propTypes.string,
        }),
      ),
      organized: propTypes.arrayOf(
        propTypes.shape({
          id: propTypes.string,
          title: propTypes.string,
          catchMessage: propTypes.string,
          place: propTypes.string,
          startedAt: propTypes.string,
        }),
      ),
    }),
    refetch: propTypes.func.isRequired,
  }).isRequired,
  tab: propTypes.string,
  user: propTypes.shape({
    fetchUser: propTypes.shape({
      uid: propTypes.string,
      displayName: propTypes.string,
      categories: propTypes.string,
      notifications: propTypes.arrayOf(propTypes.object),
    }),
  }),
  createEvent: propTypes.func.isRequired,
  authUser: propTypes.shape({
    displayName: propTypes.string.isRequired,
    uid: propTypes.string.isRequired,
  }).isRequired,
  history: propTypes.shape({
    push: propTypes.func.isRequired,
    goBack: propTypes.func.isRequired,
    replace: propTypes.func.isRequired,
  }).isRequired,
  firebase: propTypes.object.isRequired,
};

Menu.defaultProps = {
  tab: null,
};

export default withStyles(styles)(Menu);
