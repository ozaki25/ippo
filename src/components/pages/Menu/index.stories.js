import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Menu from '.';

const stories = storiesOf('pages/Menu', module);

const internalEvent = i => ({
  id: `ID${i}`,
  title: `タイトル${i}`,
  eventUrl: 'https://connpass.com/',
  catchMessage: '説明です',
  place: '晴海',
  startedAt: '2018/12/1 9:30',
});

const externalEvent = i => ({
  id: String(i),
  title: `タイトル${i}`,
  eventUrl: 'https://connpass.com/',
  catchMessage: '説明です',
  place: '晴海',
  startedAt: '2018/12/1 9:30',
});

const props = {
  createEvent: action('createEvent'),
  joined: {
    loading: false,
    joinedEvents: {
      items: [...[...Array(10)].map((_, i) => internalEvent(i))],
    },
  },
  organized: {
    loading: false,
    organizedEvents: {
      items: [...[...Array(10)].map((_, i) => internalEvent(i))],
    },
  },
  internal: {
    loading: false,
    internalEvents: {
      items: [...[...Array(10)].map((_, i) => internalEvent(i))],
    },
  },
  external: {
    loading: false,
    externalEvents: {
      items: [...[...Array(10)].map((_, i) => externalEvent(i))],
    },
  },
  authUser: {
    uid: '123',
    displayName: 'テストユーザ',
  },
  history: {
    push: action('push'),
    goBack: action('goBack'),
    replace: action('replace'),
  },
  firebase: {},
};

stories.add('通常パターン', () => <Menu {...props} />);
