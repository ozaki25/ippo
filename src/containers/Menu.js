import { graphql } from 'react-apollo';
import { compose } from 'recompose';
import { withRouter } from 'react-router-dom';
import { withFirebase } from 'context/firebase';
import query from 'graphql/query';
import mutation from 'graphql/mutation';
import Menu from 'components/pages/Menu';
import { withAuthorization } from 'hoc/Sessions';
import withTab from 'hoc/withTab';
import paging from 'constants/paging';

export default compose(
  withAuthorization,
  withRouter,
  withFirebase,
  withTab,
  graphql(query.allEvents, {
    options: ({ authUser: { uid } }) => ({
      variables: { uid, limit: paging.eventsPerPageForMenu },
    }),
  }),
  graphql(query.fetchUser, {
    options: ({ authUser: { uid } }) => ({ variables: { uid } }),
    name: 'user',
  }),
  graphql(mutation.createEvent, { name: 'createEvent' }),
)(Menu);
