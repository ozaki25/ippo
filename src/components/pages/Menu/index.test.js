import React from 'react';
import { snapshot } from 'test/helpers';
import Menu from '.';

const internalEvent = i => ({
  id: `ID${i}`,
  title: `タイトル${i}`,
  eventUrl: 'https://connpass.com/',
  catchMessage: '説明です',
  place: '晴海',
  startedAt: '2018/12/1 9:30',
});

const externalEvent = i => ({
  id: String(i),
  title: `タイトル${i}`,
  eventUrl: 'https://connpass.com/',
  catchMessage: '説明です',
  place: '晴海',
  startedAt: '2018/12/1 9:30',
});

const props = {
  createEvent: jest.fn(),
  joined: {
    loading: false,
    joinedEvents: {
      items: [...[...Array(10)].map((_, i) => internalEvent(i))],
    },
  },
  organized: {
    loading: false,
    organizedEvents: {
      items: [...[...Array(10)].map((_, i) => internalEvent(i))],
    },
  },
  internal: {
    loading: false,
    internalEvents: [...[...Array(10)].map((_, i) => internalEvent(i))],
  },
  external: {
    loading: false,
    externalEvents: {
      items: [...[...Array(10)].map((_, i) => externalEvent(i))],
    },
  },
  authUser: {
    uid: '123',
    displayName: 'テストユーザ',
  },
  history: {
    push: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
  },
  firebase: {
    doSignOut: jest.fn(),
  },
};

snapshot('Menu/nomal', <Menu {...props} />);
