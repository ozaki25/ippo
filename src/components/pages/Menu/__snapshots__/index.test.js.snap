// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Menu/nomal 1`] = `
.c1 {
  margin: 8px 0;
}

.c0 {
  padding: 10px 15px;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Menu
    authUser={null}
    external={
      Object {
        "connpass": Object {
          "events": Array [
            Object {
              "catch": "説明です",
              "event_id": 0,
              "event_url": "https://connpass.com/",
              "place": "晴海",
              "started_at": "2018/12/1 9:30",
              "title": "タイトル0",
            },
            Object {
              "catch": "説明です",
              "event_id": 1,
              "event_url": "https://connpass.com/",
              "place": "晴海",
              "started_at": "2018/12/1 9:30",
              "title": "タイトル1",
            },
            Object {
              "catch": "説明です",
              "event_id": 2,
              "event_url": "https://connpass.com/",
              "place": "晴海",
              "started_at": "2018/12/1 9:30",
              "title": "タイトル2",
            },
            Object {
              "catch": "説明です",
              "event_id": 3,
              "event_url": "https://connpass.com/",
              "place": "晴海",
              "started_at": "2018/12/1 9:30",
              "title": "タイトル3",
            },
            Object {
              "catch": "説明です",
              "event_id": 4,
              "event_url": "https://connpass.com/",
              "place": "晴海",
              "started_at": "2018/12/1 9:30",
              "title": "タイトル4",
            },
          ],
        },
        "loading": false,
      }
    }
    internal={
      Object {
        "internalEvents": Array [
          Object {
            "catchMessage": "説明です",
            "eventUrl": "https://connpass.com/",
            "id": "ID0",
            "place": "晴海",
            "startedAt": "2018/12/1 9:30",
            "title": "タイトル0",
          },
          Object {
            "catchMessage": "説明です",
            "eventUrl": "https://connpass.com/",
            "id": "ID1",
            "place": "晴海",
            "startedAt": "2018/12/1 9:30",
            "title": "タイトル1",
          },
          Object {
            "catchMessage": "説明です",
            "eventUrl": "https://connpass.com/",
            "id": "ID2",
            "place": "晴海",
            "startedAt": "2018/12/1 9:30",
            "title": "タイトル2",
          },
          Object {
            "catchMessage": "説明です",
            "eventUrl": "https://connpass.com/",
            "id": "ID3",
            "place": "晴海",
            "startedAt": "2018/12/1 9:30",
            "title": "タイトル3",
          },
          Object {
            "catchMessage": "説明です",
            "eventUrl": "https://connpass.com/",
            "id": "ID4",
            "place": "晴海",
            "startedAt": "2018/12/1 9:30",
            "title": "タイトル4",
          },
        ],
        "loading": false,
      }
    }
  >
    <Container
      authUser={null}
    >
      <withRouter()
        authUser={null}
      >
        <Route>
          <Component
            authUser={null}
            history={
              Object {
                "createHref": [Function],
                "path": "",
                "push": [Function],
                "replace": [Function],
              }
            }
            location=""
            match={
              Object {
                "params": Object {},
                "path": "",
                "url": "",
              }
            }
          >
            <NavigationBar
              authUser={null}
              firebase={null}
              history={
                Object {
                  "createHref": [Function],
                  "path": "",
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location=""
              match={
                Object {
                  "params": Object {},
                  "path": "",
                  "url": "",
                }
              }
            >
              <Blueprint3.Navbar>
                <div
                  className="bp3-navbar"
                >
                  <Blueprint3.NavbarGroup
                    align="left"
                  >
                    <div
                      className="bp3-navbar-group bp3-align-left"
                    >
                      <Blueprint3.NavbarHeading>
                        <div
                          className="bp3-navbar-heading"
                        >
                          IPPO
                        </div>
                      </Blueprint3.NavbarHeading>
                      <Blueprint3.NavbarDivider>
                        <div
                          className="bp3-navbar-divider"
                        />
                      </Blueprint3.NavbarDivider>
                    </div>
                  </Blueprint3.NavbarGroup>
                  <Blueprint3.NavbarGroup
                    align="right"
                  >
                    <div
                      className="bp3-navbar-group bp3-align-right"
                    >
                      <Blueprint3.Button
                        icon="wrench"
                        minimal={true}
                        onClick={[Function]}
                      >
                        <button
                          className="bp3-button bp3-minimal"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          type="button"
                        >
                          <Blueprint3.Icon
                            icon="wrench"
                            key="leftIcon"
                          >
                            <span
                              className="bp3-icon bp3-icon-wrench"
                              icon="wrench"
                            >
                              <svg
                                data-icon="wrench"
                                height={16}
                                viewBox="0 0 16 16"
                                width={16}
                              >
                                <desc>
                                  wrench
                                </desc>
                                <path
                                  d="M15.83 3.7l-3.06 3.05-2.84-.7-.7-2.83L12.29.17a5.004 5.004 0 0 0-4.83 1.29 4.967 4.967 0 0 0-1.12 5.36L.58 12.58c-.36.36-.58.86-.58 1.41 0 1.1.9 2 2 2 .55 0 1.05-.22 1.41-.59l5.77-5.77c1.79.69 3.91.33 5.35-1.12 1.32-1.3 1.74-3.15 1.3-4.81z"
                                  fillRule="evenodd"
                                  key="0"
                                />
                              </svg>
                            </span>
                          </Blueprint3.Icon>
                          <Blueprint3.Icon
                            key="rightIcon"
                          />
                        </button>
                      </Blueprint3.Button>
                    </div>
                  </Blueprint3.NavbarGroup>
                </div>
              </Blueprint3.Navbar>
            </NavigationBar>
          </Component>
        </Route>
      </withRouter()>
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  padding: 10px 15px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <Component>
              <h3
                className="bp3-heading"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    onClick={[Function]}
                  >
                    参加予定イベント
                  </a>
                </Link>
              </h3>
            </Component>
            <AsyncSwipeable
              loading={false}
            >
              <ShrinkSwipeable>
                <ReactSwipableView
                  animateHeight={false}
                  animateTransitions={true}
                  axis="x"
                  disableLazyLoading={false}
                  disabled={false}
                  enableMouseEvents={true}
                  hysteresis={0.4}
                  ignoreNativeScroll={false}
                  index={0}
                  resistance={false}
                  slideStyle={
                    Object {
                      "padding": "0 5px",
                    }
                  }
                  springConfig={
                    Object {
                      "delay": "0s",
                      "duration": "0.35s",
                      "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                    }
                  }
                  style={
                    Object {
                      "padding": "0 15px",
                    }
                  }
                  threshold={5}
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onScroll={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "overflowX": "hidden",
                        "padding": "0 15px",
                      }
                    }
                  >
                    <div
                      className="react-swipeable-view-container"
                      style={
                        Object {
                          "WebkitFlexDirection": "row",
                          "WebkitTransition": "all 0s ease 0s",
                          "direction": "ltr",
                          "display": "flex",
                          "flexDirection": "row",
                          "height": null,
                          "transition": "all 0s ease 0s",
                          "willChange": "transform",
                        }
                      }
                    >
                      <div
                        aria-hidden={false}
                        data-swipeable="true"
                        key=".$ID0"
                        style={
                          Object {
                            "WebkitFlexShrink": 0,
                            "flexShrink": 0,
                            "overflow": "auto",
                            "padding": "0 5px",
                            "width": "100%",
                          }
                        }
                      >
                        <styled.div
                          key="ID0"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bdVaJa",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "
  margin: 8px 0;
",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bdVaJa",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c1"
                            >
                              <EventCard
                                catchMessage="説明です"
                                datetime="2018年12月01日 09:30 〜"
                                eventUrl="https://connpass.com/"
                                interactive={true}
                                place="晴海"
                                title="タイトル0"
                              >
                                <Blueprint3.Card
                                  elevation={2}
                                  interactive={true}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="bp3-card bp3-interactive bp3-elevation-2"
                                    onClick={[Function]}
                                  >
                                    <h3
                                      className="bp3-heading"
                                    >
                                      タイトル0
                                    </h3>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        概要：説明です
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        場所：晴海
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        日時：2018年12月01日 09:30 〜
                                      </div>
                                    </Blueprint3.Text>
                                  </div>
                                </Blueprint3.Card>
                              </EventCard>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </div>
                  </div>
                </ReactSwipableView>
              </ShrinkSwipeable>
            </AsyncSwipeable>
            <Component>
              <h3
                className="bp3-heading"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  <a
                    onClick={[Function]}
                  >
                    おすすめイベント
                  </a>
                </Link>
              </h3>
            </Component>
            <AsyncSwipeable
              loading={false}
            >
              <ShrinkSwipeable>
                <ReactSwipableView
                  animateHeight={false}
                  animateTransitions={true}
                  axis="x"
                  disableLazyLoading={false}
                  disabled={false}
                  enableMouseEvents={true}
                  hysteresis={0.4}
                  ignoreNativeScroll={false}
                  index={0}
                  resistance={false}
                  slideStyle={
                    Object {
                      "padding": "0 5px",
                    }
                  }
                  springConfig={
                    Object {
                      "delay": "0s",
                      "duration": "0.35s",
                      "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                    }
                  }
                  style={
                    Object {
                      "padding": "0 15px",
                    }
                  }
                  threshold={5}
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onScroll={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "overflowX": "hidden",
                        "padding": "0 15px",
                      }
                    }
                  >
                    <div
                      className="react-swipeable-view-container"
                      style={
                        Object {
                          "WebkitFlexDirection": "row",
                          "WebkitTransition": "all 0s ease 0s",
                          "direction": "ltr",
                          "display": "flex",
                          "flexDirection": "row",
                          "height": null,
                          "transition": "all 0s ease 0s",
                          "willChange": "transform",
                        }
                      }
                    >
                      <div
                        aria-hidden={false}
                        data-swipeable="true"
                        key=".$ID0"
                        style={
                          Object {
                            "WebkitFlexShrink": 0,
                            "flexShrink": 0,
                            "overflow": "auto",
                            "padding": "0 5px",
                            "width": "100%",
                          }
                        }
                      >
                        <styled.div
                          key="ID0"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bdVaJa",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "
  margin: 8px 0;
",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bdVaJa",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c1"
                            >
                              <EventCard
                                catchMessage="説明です"
                                datetime="2018年12月01日 09:30 〜"
                                eventUrl="https://connpass.com/"
                                interactive={true}
                                place="晴海"
                                title="タイトル0"
                              >
                                <Blueprint3.Card
                                  elevation={2}
                                  interactive={true}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="bp3-card bp3-interactive bp3-elevation-2"
                                    onClick={[Function]}
                                  >
                                    <h3
                                      className="bp3-heading"
                                    >
                                      タイトル0
                                    </h3>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        概要：説明です
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        場所：晴海
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        日時：2018年12月01日 09:30 〜
                                      </div>
                                    </Blueprint3.Text>
                                  </div>
                                </Blueprint3.Card>
                              </EventCard>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </div>
                  </div>
                </ReactSwipableView>
              </ShrinkSwipeable>
            </AsyncSwipeable>
            <Component>
              <h3
                className="bp3-heading"
              >
                <Link
                  replace={false}
                  to="/events/internal"
                >
                  <a
                    onClick={[Function]}
                  >
                    社内イベント
                  </a>
                </Link>
              </h3>
            </Component>
            <AsyncSwipeable
              loading={false}
            >
              <ShrinkSwipeable>
                <ReactSwipableView
                  animateHeight={false}
                  animateTransitions={true}
                  axis="x"
                  disableLazyLoading={false}
                  disabled={false}
                  enableMouseEvents={true}
                  hysteresis={0.4}
                  ignoreNativeScroll={false}
                  index={0}
                  resistance={false}
                  slideStyle={
                    Object {
                      "padding": "0 5px",
                    }
                  }
                  springConfig={
                    Object {
                      "delay": "0s",
                      "duration": "0.35s",
                      "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                    }
                  }
                  style={
                    Object {
                      "padding": "0 15px",
                    }
                  }
                  threshold={5}
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onScroll={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "overflowX": "hidden",
                        "padding": "0 15px",
                      }
                    }
                  >
                    <div
                      className="react-swipeable-view-container"
                      style={
                        Object {
                          "WebkitFlexDirection": "row",
                          "WebkitTransition": "all 0s ease 0s",
                          "direction": "ltr",
                          "display": "flex",
                          "flexDirection": "row",
                          "height": null,
                          "transition": "all 0s ease 0s",
                          "willChange": "transform",
                        }
                      }
                    >
                      <div
                        aria-hidden={false}
                        data-swipeable="true"
                        key=".$ID0"
                        style={
                          Object {
                            "WebkitFlexShrink": 0,
                            "flexShrink": 0,
                            "overflow": "auto",
                            "padding": "0 5px",
                            "width": "100%",
                          }
                        }
                      >
                        <styled.div
                          key="ID0"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bdVaJa",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "
  margin: 8px 0;
",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bdVaJa",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c1"
                            >
                              <EventCard
                                catchMessage="説明です"
                                datetime="2018年12月01日 09:30 〜"
                                eventUrl="https://connpass.com/"
                                interactive={true}
                                place="晴海"
                                title="タイトル0"
                              >
                                <Blueprint3.Card
                                  elevation={2}
                                  interactive={true}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="bp3-card bp3-interactive bp3-elevation-2"
                                    onClick={[Function]}
                                  >
                                    <h3
                                      className="bp3-heading"
                                    >
                                      タイトル0
                                    </h3>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        概要：説明です
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        場所：晴海
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        日時：2018年12月01日 09:30 〜
                                      </div>
                                    </Blueprint3.Text>
                                  </div>
                                </Blueprint3.Card>
                              </EventCard>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </div>
                  </div>
                </ReactSwipableView>
              </ShrinkSwipeable>
            </AsyncSwipeable>
            <Component>
              <h3
                className="bp3-heading"
              >
                <Link
                  replace={false}
                  to="/events/external"
                >
                  <a
                    onClick={[Function]}
                  >
                    社外イベント
                  </a>
                </Link>
              </h3>
            </Component>
            <AsyncSwipeable
              loading={false}
            >
              <ShrinkSwipeable>
                <ReactSwipableView
                  animateHeight={false}
                  animateTransitions={true}
                  axis="x"
                  disableLazyLoading={false}
                  disabled={false}
                  enableMouseEvents={true}
                  hysteresis={0.4}
                  ignoreNativeScroll={false}
                  index={0}
                  resistance={false}
                  slideStyle={
                    Object {
                      "padding": "0 5px",
                    }
                  }
                  springConfig={
                    Object {
                      "delay": "0s",
                      "duration": "0.35s",
                      "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                    }
                  }
                  style={
                    Object {
                      "padding": "0 15px",
                    }
                  }
                  threshold={5}
                >
                  <div
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    onMouseUp={[Function]}
                    onScroll={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "overflowX": "hidden",
                        "padding": "0 15px",
                      }
                    }
                  >
                    <div
                      className="react-swipeable-view-container"
                      style={
                        Object {
                          "WebkitFlexDirection": "row",
                          "WebkitTransition": "all 0s ease 0s",
                          "direction": "ltr",
                          "display": "flex",
                          "flexDirection": "row",
                          "height": null,
                          "transition": "all 0s ease 0s",
                          "willChange": "transform",
                        }
                      }
                    >
                      <div
                        aria-hidden={false}
                        data-swipeable="true"
                        key=".$0"
                        style={
                          Object {
                            "WebkitFlexShrink": 0,
                            "flexShrink": 0,
                            "overflow": "auto",
                            "padding": "0 5px",
                            "width": "100%",
                          }
                        }
                      >
                        <styled.div
                          key="0"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-bdVaJa",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "
  margin: 8px 0;
",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-bdVaJa",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="c1"
                            >
                              <EventCard
                                catchMessage="説明です"
                                datetime="2018年12月01日 09:30 〜"
                                eventUrl="https://connpass.com/"
                                interactive={true}
                                place="晴海"
                                title="タイトル0"
                              >
                                <Blueprint3.Card
                                  elevation={2}
                                  interactive={true}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="bp3-card bp3-interactive bp3-elevation-2"
                                    onClick={[Function]}
                                  >
                                    <h3
                                      className="bp3-heading"
                                    >
                                      タイトル0
                                    </h3>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        概要：説明です
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        場所：晴海
                                      </div>
                                    </Blueprint3.Text>
                                    <Blueprint3.Text>
                                      <div
                                        className=""
                                      >
                                        日時：2018年12月01日 09:30 〜
                                      </div>
                                    </Blueprint3.Text>
                                  </div>
                                </Blueprint3.Card>
                              </EventCard>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </div>
                  </div>
                </ReactSwipableView>
              </ShrinkSwipeable>
            </AsyncSwipeable>
          </div>
        </StyledComponent>
      </styled.div>
    </Container>
  </Menu>
</MockRouter>
`;
